### ISSUE TITLE
Driver error reading a single line GeoJSONSeq file

### ISSUE THREAD

**Issue by dazza-codes** [Toblerity/Fiona#991]
fiona should be able to read a GeoJSONSeq file with only one line (one feature).

```
$ cat tmp.geojsons 
{"type": "Feature", "id": "669", "geometry": {"type": "Point", "coordinates": [170.50251, -45.874003]}, "properties": {"cityName": "Dunedin", "countryName": "New Zealand", "s2_cell_id": "a82eac0daa4c352b", "uuid": "9fbadb10-6019-41e1-9f6f-a3ddd9c6bff9"}}

$ jsonlint-php tmp.geojsons 
Valid JSON (tmp.geojsons)

$ jq '.' tmp.geojsons 
{
  "type": "Feature",
  "id": "669",
  "geometry": {
    "type": "Point",
    "coordinates": [
      170.50251,
      -45.874003
    ]
  },
  "properties": {
    "cityName": "Dunedin",
    "countryName": "New Zealand",
    "s2_cell_id": "a82eac0daa4c352b",
    "uuid": "9fbadb10-6019-41e1-9f6f-a3ddd9c6bff9"
  }
}

$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.5 LTS"

$ python
Python 3.7.8 | packaged by conda-forge | (default, Nov 27 2020, 19:24:58) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.

>>> fiona.__version__
'1.8.18'
>>> fiona.gdal_version
GDALVersion(major=2, minor=4, revision=4)
>>> fiona.supported_drivers
{'AeronavFAA': 'r', 'ARCGEN': 'r', 'BNA': 'rw', 'DXF': 'rw', 'CSV': 'raw', 'OpenFileGDB': 'r', 'ESRIJSON': 'r', 'ESRI Shapefile': 'raw', 'GeoJSON': 'raw', 'GeoJSONSeq': 'rw', 'GPKG': 'raw', 'GML': 'rw', 'OGR_GMT': 'rw', 'GPX': 'rw', 'GPSTrackMaker': 'rw', 'Idrisi': 'r', 'MapInfo File': 'raw', 'DGN': 'raw', 'OGR_PDS': 'r', 'S57': 'r', 'SEGY': 'r', 'SUA': 'r', 'TopoJSON': 'r'}
>>> fiona.supported_drivers["GeoJSONSeq"]
'rw'

>>> import geopandas as gpd
>>> gpd.__version__
'0.8.1'
>>> gdf = gpd.read_file("tmp.geojsons", driver="GeoJSONSeq")
Traceback (most recent call last):
  File "fiona/_shim.pyx", line 74, in fiona._shim.gdal_open_vector
  File "fiona/_err.pyx", line 291, in fiona._err.exc_wrap_pointer
fiona._err.CPLE_OpenFailedError: 'tmp.geojsons' not recognized as a supported file format.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/envs/project/lib/python3.7/site-packages/geopandas/io/file.py", line 96, in _read_file
    with reader(path_or_bytes, **kwargs) as features:
  File "/opt/conda/envs/project/lib/python3.7/site-packages/fiona/env.py", line 400, in wrapper
    return f(*args, **kwargs)
  File "/opt/conda/envs/project/lib/python3.7/site-packages/fiona/__init__.py", line 257, in open
    layer=layer, enabled_drivers=enabled_drivers, **kwargs)
  File "/opt/conda/envs/project/lib/python3.7/site-packages/fiona/collection.py", line 162, in __init__
    self.session.start(self, **kwargs)
  File "fiona/ogrext.pyx", line 540, in fiona.ogrext.Session.start
  File "fiona/_shim.pyx", line 81, in fiona._shim.gdal_open_vector
fiona.errors.DriverError: 'tmp.geojsons' not recognized as a supported file format.

```

**Comment by dazza-codes** [comment]
The `fio` CLI seems to read it fine:

```
$ fio insp tmp.geojsons 
Fiona 1.8.18 Interactive Inspector (Python 3.7.8)
Type "src.schema", "next(src)", or "help(src)" for more information.
>>> src
<open Collection 'tmp.geojsons:tmp', mode 'r' at 0x7f2e76990750>
>>> dir(src)
['__class__', '__contains__', '__del__', '__delattr__', '__dict__', '__dir__', '__doc__', '__enter__', '__eq__', '__exit__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__iter__', '__le__', '__len__', '__lt__', '__module__', '__ne__', '__new__', '__next__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_bounds', '_check_schema_driver_support', '_crs', '_crs_wkt', '_driver', '_env', '_len', '_schema', 'bounds', 'close', 'closed', 'crs', 'crs_wkt', 'driver', 'enabled_drivers', 'encoding', 'env', 'field_skip_log_filter', 'filter', 'flush', 'get', 'guard_driver_mode', 'ignore_fields', 'ignore_geometry', 'items', 'iterator', 'keys', 'meta', 'mode', 'name', 'next', 'path', 'profile', 'schema', 'session', 'validate_record', 'validate_record_geometry', 'values', 'write', 'writerecords']
>>> src.enabled_drivers
>>> src.enabled_drivers is None
True
>>> src.items()
<fiona.ogrext.ItemsIterator object at 0x7f2e76ea4ad0>
>>> list(src.items())
[(0, {'type': 'Feature', 'id': '0', 'properties': OrderedDict([('id', '669'), ('cityName', 'Dunedin'), ('countryName', 'New Zealand'), ('s2_cell_id', 'a82eac0daa4c352b'), ('uuid', '9fbadb10-6019-41e1-9f6f-a3ddd9c6bff9')]), 'geometry': {'type': 'Point', 'coordinates': (170.50251, -45.874003)}})]
>>> 
>>> features = list(src.values())
>>> features
[{'type': 'Feature', 'id': '0', 'properties': OrderedDict([('id', '669'), ('cityName', 'Dunedin'), ('countryName', 'New Zealand'), ('s2_cell_id', 'a82eac0daa4c352b'), ('uuid', '9fbadb10-6019-41e1-9f6f-a3ddd9c6bff9')]), 'geometry': {'type': 'Point', 'coordinates': (170.50251, -45.874003)}}]
```

What's the difference between the python repl and a simple `import fiona` and the `fio` that explains this discrepancy in how fiona reads this file?

**Comment by sgillies** [comment]
In the first example you're using gpd.read_file. Different fiona installs perhaps?

I found https://github.com/OSGeo/gdal/blob/master/gdal/NEWS#L1109, which indicates to me that this is a GDAL bug in 2.4.4 that has been fixed in 3.1.0. Can you verify that?
